networklibrary/src/main/java/com/eis/smsnetwork/broadcast/BroadcastReceiver.java

```diff
@@ -0,0 +1,182 @@
+     * The rest of the message varies depending on the {@link RequestType}:
+     * <ul>
+     * <li>Invite: there are no other fields</li>
+     * <li>AcceptInvitation: there are no other fields</li>
+     * <li>AddPeer: fields from 1 to the last one contain the phone numbers of each
+     * {@link com.eis.smslibrary.SMSPeer} we have to add to our network</li>
+     * <li>QuitNetwork: there are no other fields, because this request can only be sent by the
+     * {@link com.eis.smslibrary.SMSPeer} who wants to be removed</li>
+     * <li>AddResource: starting from 1, fields with odd numbers contain keys, their following
+     * (even)
+     * field contains the corresponding value</li>
+     * <li>RemoveResource: fields from 1 to the last one contain the keys to remove</li>
+     * </ul>
+     *
+     * @param message The message passed by {@link com.eis.smslibrary.SMSReceivedBroadcastReceiver}.
+     */
+    @Override
+    public void onMessageReceived(SMSMessage message) {
+        Log.d("BR_RECEIVER", "Message received: " + message.getPeer() + " " + message.getData());
+        String[] fields = message.getData().split(SEPARATOR_REGEX);
+        RequestType request;
+        try {
+            request = RequestType.get(fields[0]);
+        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+            return;
+        }
+        if (request == null) return;
+        SMSPeer sender = message.getPeer();
+        SMSNetSubscriberList subscribers =
+                (SMSNetSubscriberList) SMSJoinableNetManager.getInstance().getNetSubscriberList();
+        SMSNetDictionary dictionary =
+                (SMSNetDictionary) SMSJoinableNetManager.getInstance().getNetDictionary();
+        boolean senderIsNotSubscriber = !subscribers.getSubscribers().contains(sender);
+
+        switch (request) {
+            case Invite: {
+                if (fields.length > 1) return;
+                SMSJoinableNetManager.getInstance().checkInvitation(new SMSInvitation(sender));
+                break;
+            }
+            case AcceptInvitation: {
+                if (fields.length > 1) return;
+                // Verifying if the sender has been invited to join the network
+                Set<SMSPeer> invitedPeers = SMSJoinableNetManager.getInstance().getInvitedPeers();
+                if (!invitedPeers.contains(sender))
+                    return;
+                invitedPeers.remove(sender);
+
+                // Sending to the invited peer my subscribers list
+                StringBuilder mySubscribers = new StringBuilder(RequestType.AddPeer.asString() +
+                        FIELD_SEPARATOR);
+                for (SMSPeer peerToAdd : subscribers.getSubscribers())
+                    mySubscribers.append(peerToAdd).append(FIELD_SEPARATOR);
+                SMSMessage mySubscribersMessage = new SMSMessage(
+                        sender, mySubscribers.deleteCharAt(mySubscribers.length() - 1).toString());
```

Comment:
Although it's evident from the code, I would have explained with a comment why do you remove the last char of the `mySubscribers` string to make this code more "readable"

Reply:
Ok, I'll do it